packages/dynamic_layouts/lib/src/render_dynamic_grid.dart

```diff
@@ -0,0 +1,410 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'base_grid_layout.dart';
+
+/// A sliver that places multiple box children in a two dimensional arrangement.
+///
+/// [RenderDynamicSliverGrid] places its children in arbitrary positions determined by
+/// the [DynamicSliverGridLayout] provided by the [gridDelegate].
+class RenderDynamicSliverGrid extends RenderSliverMultiBoxAdaptor {
+  /// Creates a sliver that contains multiple box children that whose size and
+  /// position are determined by a delegate.
+  ///
+  /// The [childManager] and [gridDelegate] arguments must not be null.
+  RenderDynamicSliverGrid({
+    required super.childManager,
+    required SliverGridDelegate gridDelegate,
+  }) : assert(gridDelegate != null),
+       _gridDelegate = gridDelegate;
+
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverGridParentData) {
+      child.parentData = SliverGridParentData();
+    }
+  }
+
+  /// The delegate that controls the size and position of the children.
+  SliverGridDelegate get gridDelegate => _gridDelegate;
+  SliverGridDelegate _gridDelegate;
+  set gridDelegate(SliverGridDelegate value) {
+    assert(value != null);
+    if (_gridDelegate == value) {
+      return;
+    }
+    if (value.runtimeType != _gridDelegate.runtimeType ||
+        value.shouldRelayout(_gridDelegate)) {
+      markNeedsLayout();
+    }
+    _gridDelegate = value;
+  }
+
+  @override
+  double childCrossAxisPosition(RenderBox child) {
+    final SliverGridParentData childParentData = child.parentData! as SliverGridParentData;
+    return childParentData.crossAxisOffset!;
+  }
+
+  @override
+  void performLayout() {
```

Comment:
Is a lot of this logic shared with e.g. RenderSliverGrid? If true, wondering, if we could find a way to share that better. We don't have to necessarily do it in this PR, though. But maybe add a TODO if you think it is possible...

Reply:
performLayout is really where it differs from RenderDynamicSliverGrid, I'll clean it up a bit.